### PR TITLE
Phase2-hgx223 Update the validity finding code

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Correct the code which decides the valid detid of silicon cells

#### PR validation:

Use standard runTheMatrix code to check the changes

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -1495,6 +1473,39 @@ bool HGCalDDDConstants::waferInLayerTest(int wafer, int lay, bool full) const {
   return in;
 }
 
+std::pair<double, double> HGCalDDDConstants::waferPosition(int lay,
+                                                           int waferU,
+                                                           int waferV,
+                                                           bool reco,
+                                                           bool
+#ifdef EDM_ML_DEBUG
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: is this really necessary?

**Comment by kpedro88** [comment]
are comparison changes expected here? if so, what causes the changes?

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 1 above)
Comment (fabiocos) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: it looks a similar situation to another update of a few days ago, where we preferred at the end not to have the argument list conditional to a debug flag

**Comment by bsunanda** [comment]
Earlier the validity of a cell was incomplete. So some silicon cells were declared valid though they were not. This affects cells at the edges in the so called virtual wafers. So some difference is expected.

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: Since debug is enclosed within ifdef, clang complains of unused variable

Comment (kpedro88) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: Is the debug flag really needed in addition to the ifdef?

Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: I can remove ifdef part for this code. "debug" is needed for some special debugging

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**